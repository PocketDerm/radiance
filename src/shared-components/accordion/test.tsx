import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Accordion from './index';

const testAccordion = {
  title: <div>title</div>,
  isOpen: false,
  onClick: (): void => undefined,
  children: <div>expansion</div>,
};

/* eslint-disable react/jsx-props-no-spreading */
describe('<Accordion />', () => {
  test('renders regular accordion', () => {
    const component = renderer.create(<Accordion {...testAccordion} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders no border accordion', () => {
    const component = renderer.create(
      <Accordion noBorder {...testAccordion} />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders disabled accordion', () => {
    const component = renderer.create(
      <Accordion disabled {...testAccordion} />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('invokes onClick when title is clicked', () => {
    const spy = jest.fn();
    const component = shallow(<Accordion {...testAccordion} onClick={spy} />);

    component.find('[role="button"]').simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
/* eslint-enable react/jsx-props-no-spreading */
