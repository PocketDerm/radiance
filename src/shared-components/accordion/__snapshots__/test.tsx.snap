// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders disabled accordion 1`] = `
.emotion-0 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
}

.emotion-6 {
  max-height: 0px;
  overflow: hidden;
  -webkit-transition: max-height 350ms ease-in-out;
  transition: max-height 350ms ease-in-out;
}

.emotion-8 {
  border-radius: 4px;
  border: 1px solid #ededf0;
  width: 100%;
  opacity: 0.4;
  background-color: #ededf0;
  border-color: #c3c0cd;
}

.emotion-8:not(:last-child) {
  border-bottom: none;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: not-allowed;
}

.emotion-4:focus {
  outline: none;
  box-shadow: 0 0 2px 2px rgba(166,161,226,0.5);
}

<div
  className="emotion-8 emotion-9"
  disabled={true}
>
  <div
    aria-disabled={true}
    aria-expanded={false}
    className="emotion-4 emotion-5"
    disabled={true}
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={-1}
  >
    <div
      className="emotion-0 emotion-1"
    >
      <div>
        title
      </div>
    </div>
    <div
      className="emotion-2 emotion-3"
    >
      <svg
        fill="#332e54"
        height={16}
        rotate={0}
        width={16}
      />
    </div>
  </div>
  <div
    aria-disabled={true}
    aria-hidden={true}
    className="emotion-6 emotion-7"
  >
    <div>
      <div>
        expansion
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders no border accordion 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
}

.emotion-4:focus {
  outline: none;
  box-shadow: 0 0 2px 2px rgba(166,161,226,0.5);
}

.emotion-0 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
}

.emotion-6 {
  max-height: 0px;
  overflow: hidden;
  -webkit-transition: max-height 350ms ease-in-out;
  transition: max-height 350ms ease-in-out;
}

.emotion-8 {
  border-radius: 4px;
  width: 100%;
}

.emotion-8:not(:last-child) {
  border-bottom: none;
}

<div
  className="emotion-8 emotion-9"
  disabled={false}
>
  <div
    aria-disabled={false}
    aria-expanded={false}
    className="emotion-4 emotion-5"
    disabled={false}
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="emotion-0 emotion-1"
    >
      <div>
        title
      </div>
    </div>
    <div
      className="emotion-2 emotion-3"
    >
      <svg
        fill="#332e54"
        height={16}
        rotate={0}
        width={16}
      />
    </div>
  </div>
  <div
    aria-disabled={false}
    aria-hidden={true}
    className="emotion-6 emotion-7"
  >
    <div>
      <div>
        expansion
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders regular accordion 1`] = `
.emotion-8 {
  border-radius: 4px;
  border: 1px solid #ededf0;
  width: 100%;
}

.emotion-8:not(:last-child) {
  border-bottom: none;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
}

.emotion-4:focus {
  outline: none;
  box-shadow: 0 0 2px 2px rgba(166,161,226,0.5);
}

.emotion-0 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
}

.emotion-6 {
  max-height: 0px;
  overflow: hidden;
  -webkit-transition: max-height 350ms ease-in-out;
  transition: max-height 350ms ease-in-out;
}

<div
  className="emotion-8 emotion-9"
  disabled={false}
>
  <div
    aria-disabled={false}
    aria-expanded={false}
    className="emotion-4 emotion-5"
    disabled={false}
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="emotion-0 emotion-1"
    >
      <div>
        title
      </div>
    </div>
    <div
      className="emotion-2 emotion-3"
    >
      <svg
        fill="#332e54"
        height={16}
        rotate={0}
        width={16}
      />
    </div>
  </div>
  <div
    aria-disabled={false}
    aria-hidden={true}
    className="emotion-6 emotion-7"
  >
    <div>
      <div>
        expansion
      </div>
    </div>
  </div>
</div>
`;
